mypy/checkmember.py

```diff
@@ -642,8 +642,13 @@ class B(A): pass
             # XXX value restriction as union?
             original_type = erase_to_bound(self_param_type)
 
-        typearg = infer_type_arguments([x.id for x in func.variables],
-                                       self_param_type, original_type)[0]
+        ids = [x.id for x in func.variables]
+        typearg = infer_type_arguments(ids, self_param_type, original_type)[0]
+        if (is_classmethod and isinstance(typearg, UninhabitedType)
```

Comment:
OK, this seems reasonable. If there will be no objections, then I am going to merge this.